text/0026-semantic-versioning.md

```diff
@@ -0,0 +1,189 @@
+
+As a Buildpack/Builder maintainer,<br/>
+I need a release versioning guideline,
+so I can make my release versions consistent.
+
+As a Buildpack/Builder user,<br/>
+I want to know how versioning is done,
+so I can use the appropriate version and
+decide when to upgrade.
+
+## Semantics
+
+Semantic Versioning 2.0.0 defines versioning syntax as follows ([semver.org](https://semver.org/)):
+
+> Given a version number MAJOR.MINOR.PATCH, increment the:
+> 
+>   - MAJOR version when you make incompatible API changes,
+>   - MINOR version when you add functionality in a backwards compatible manner, and
+>   - PATCH version when you make backwards compatible bug fixes.
+
+The proposal is to conform to the above semantics as much as possible.
+
+### Changes
+
+In terms of Paketo Buildpacks/Builders, this translates to:
+
+* Bumping MAJOR is analogous to an "`app or related buildpacks is likely to not
+	work without change`" relation between releases.
+* Bumping MINOR is analogous to an "`updates or adds features`" relation
+	between releases.
+* Bumping PATCH is analogous to a "`fixes issue`" relation between releases.
+
+The following table lists the usual circumstances/events where bumping a particular component is expected:
+
+#### Buildpack (General)
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| \<Any incompatible change\> | \<Feature additions but compatible with previous release\> | \<Fixes but no major feature changes\> |
+| Change in Buildpack ID | Support for a new stack | Bug fixes |
+| Change in Buildpack API version | Buildpack runs a new command the causes backward compatible<br>change in behavior (e.g. x-install buildpack adds running "x clean") | Performance improvement, internal tweaks |
+| Change in API (Provides/Requires) | New configuration option (e.g. new environment variable support) | CVE fixes |
+| Removal of a stack or renaming of an existing stack | Change in <br>[buildpack-specific metadata](https://github.com/buildpacks/spec/blob/buildpack/v0.6/buildpack.md#build-plan-toml) | Change of required build tool version to package buildpack<br>(e.g. go version) |
+| Removal of configuration options<br>(e.g. buildpack.yml) |  |  |
+<br/>
+
+#### Buildpack Dependencies
+
+For buildpacks whose primary function is to provide dependencies, the following versioning is expectated
+when versions of dependencies are added or removed. The table below assumes that upstream dependencies
+follow [semantic versioning](https://semver.org/). For those that do not, maintainers may determine
+their versioning independently.
+
+| Dependency change | Buildpack Version expectation |
+|-|-|
+| ADD Patch | Bump Patch |
+| ADD Minor | - Bump Minor if set as default<br>  version<br>- Bump Patch otherwise |
+| ADD Major | - Bump Major if set as default<br>  version<br>- Bump Minor otherwise |
+| REMOVE Patch | Bump Patch |
+| REMOVE Minor | - Bump Major if unset as default<br>  version<br>- Bump Minor otherwise |
+| REMOVE Major | Bump Major |
+<br/>
```

Comment:
I am finding the add/remove+default semantics a bit hard to follow. Maybe we should separating the defaulting behavior from the add/remove. i.e.
- Add
  - patch -> patch
  - minor -> minor
  - major -> minor
Remove
  - patch-> patch
  - minor -> minor
  - major -> major
- change default:
   - patch (e.g. 1.0.0 -> 1.0.1) -> patch
   - minor (e.g. 1.0.1 -> 1.1.0) -> minor
   - major (e.g. 1.1.0 -> 2.0.0) -> major

Sometimes I do two things at the same time (e.g. replace a default dependency with the next patch version of the same dependency). But I could also theoretically change the default without adding or removing anything.

Reply:
I see a similar case addressed at https://github.com/paketo-buildpacks/rfcs/pull/49#discussion_r615011293. I'll separate them out.